spring-integration-core/src/main/java/org/springframework/integration/endpoint/EventDrivenConsumer.java

```diff
@@ -58,6 +59,9 @@ public MessageChannel getOutputChannel() {
 		if (this.handler instanceof MessageProducer) {
 			return ((MessageProducer) this.handler).getOutputChannel();
 		}
+		else if (this.handler instanceof AbstractMessageRouter) {
+			return ((AbstractMessageRouter) this.handler).getDefaultOutputChannel();
```

Comment:
I wonder if it may be proxy and therefore we need one more special interface...
WDYT?